File: 843/843_version1.java
/**
 * // This is the Master's API interface.
 * // You should not implement it, or speculate about its implementation
 * interface Master {
 *     public int guess(String word) {}
 * }
 */
class Solution {
    public void findSecretWord(String[] wordlist, Master master) {
        Random seed = new Random();
        for(int time = 0,match = 0;time < 10 && match < 6;time++){
            String candidate = wordlist[seed.nextInt(wordlist.length)];
            List<String> newWordList = new ArrayList<>();
            match = master.guess(candidate);
            for(String w:wordlist) 
                if(compareWords(w,candidate)==match)
                    newWordList.add(w);
            wordlist = newWordList.toArray(new String[newWordList.size()]);
        }
    }
    
    
    private int compareWords(String a,String b){
        int res = 0;
        for(int i = 0;i<6;i++){
            if(a.charAt(i) == b.charAt(i)) res++;
        }
        return res;
    }
}

File: 843/843_version1.py
# """
# This is Master's API interface.
# You should not implement it, or speculate about its implementation
# """
#class Master:
#    def guess(self, word):
#        """
#        :type word: str
#        :rtype int
#        """

class Solution:
    def findSecretWord(self, wordlist, master):
        """
        :type wordlist: List[Str]
        :type master: Master
        :rtype: None
        """
        time,match = 0,0
        while time < 10 and match < 6:
            candidate = wordlist[random.randint(0,len(wordlist)-1)]
            match = master.guess(candidate)
            wordlist = [i for i in wordlist if self.compareWords(i,candidate)==match]
            time+=1
    
    def compareWords(self,word1,word2):
        res = 0
        for i in range(6):
            if word1[i] == word2[i]:
                res+=1
        return res
